src/Land/Model/soil_model.jl

```diff
@@ -0,0 +1,251 @@
+#### Soil model
+
+export SoilModel
+
+"""
+    SoilModel{W, H} <: BalanceLaw
+
+A BalanceLaw for soil modeling.
+Users may over-ride prescribed default values for each field.
+
+# Usage
+
+    SoilModel(
+        water,
+        heat,
+    )
+
```

Comment:
```suggestion
    SoilModel{W, H} <: BalanceLaw
```
The properties are already expanded by `$(DocStringExtensions.FIELDS)` (see [here](https://clima.github.io/ClimateMachine.jl/previews/PR1279/APIs/Land/LandModel/#ClimateMachine.Land.SoilModel)), so including signature/type parameters I think provides better information here.